lib/presentation/common/custom_text_form.dart

```diff
@@ -32,15 +32,17 @@ class _CustomTextFormState extends State<CustomTextForm> {
       obscureText: widget.isNonPasswordField ? false : !obscureText,
       validator: widget.validator,
       decoration: InputDecoration(
-        suffixIcon: IconButton(
-            icon: widget.isNonPasswordField
-                ? Icon(null)
-                : !obscureText
-                    ? Icon(Icons.visibility)
-                    : Icon(Icons.visibility_off),
-            onPressed: () {
-              toggleObscureText();
-            }),
+        suffixIcon: !widget.isNonPasswordField
```

Comment:
create suffiwWidget as optional param instead of this

Reply:
the logic for the suffix icon should be outside of this class. So you forward suffixIcon or not? This generic widget should not do the logic. It may happen that this widget is not password input and you still need to set suffixIcon for some reason.